src/crimson/os/seastore/btree/fixed_kv_btree.h

```diff
@@ -439,6 +485,152 @@ class FixedKVBtree {
     return upper_bound(c, min_max_t<node_key_t>::max);
   }
 
+  template <typename child_node_t, typename node_t>
+  void check_node(
+    op_context_t<node_key_t> c,
+    TCachedExtentRef<node_t> node)
+  {
+    for (auto i : *node) {
+      CachedExtentRef child_node;
+      Transaction::get_extent_ret ret;
+
+      if constexpr (std::is_base_of_v<typename internal_node_t::base_t, child_node_t>) {
+        ret = c.trans.get_extent(
+          i->get_val().maybe_relative_to(node->get_paddr()),
+          &child_node);
+      } else {
+        if constexpr (leaf_has_children) {
+          ret = c.trans.get_extent(
+            i->get_val().paddr.maybe_relative_to(node->get_paddr()),
+            &child_node);
+        }
+      }
+      if (ret == Transaction::get_extent_ret::PRESENT) {
+        if (child_node->is_mutation_pending()) {
+          auto &prior = (child_node_t &)*child_node->prior_instance;
+          assert(prior.is_valid());
+          assert(prior.is_parent_valid());
+          if (node->is_mutation_pending()) {
+            auto &n = node->get_stable_for_key(i->get_key());
+            assert(prior.get_parent_node().get() == &n);
+            auto pos = n.lower_bound_offset(i->get_key());
+            assert(pos < n.get_node_size());
+            assert(n.children[pos] == &prior);
+          } else {
+            assert(prior.get_parent_node().get() == node.get());
+            assert(node->children[i->get_offset()] == &prior);
+          }
+        } else if (child_node->is_initial_pending()) {
+          auto cnode = child_node->template cast<child_node_t>();
+          auto pos = node->find(i->get_key()).get_offset();
+          auto child = node->children[pos];
+          assert(child);
+          assert(child == cnode.get());
+          assert(cnode->is_parent_valid());
+        } else {
```

Comment:
EXIST_CLEAN and EXIST_MUTATION_PENDING `child_node` seem to be falling into this else case, which is wrong because the else case expects the `child_node` is stable.

This issue was observed during testing https://github.com/ceph/ceph/pull/51179 which depends on the EXIST_CLEAN/PENDING features.

Probably need to scan other places as well to make sure EXIST_CLEAN and EXIST_MUTATION_PENDING extents are correctly handled with the lba tree pointers.

Reply:
see https://github.com/ceph/ceph/pull/51980